prime-router/src/main/kotlin/cli/tests/TestReportStream.kt

```diff
@@ -911,26 +991,68 @@ abstract class CoolTest {
                         // Get errors and warnings
                         ret.logs = ctx.selectFrom(ACTION_LOG).where(
                             ACTION_LOG.ACTION_ID.eq(report.actionId)
-                                .and(ACTION_LOG.REPORT_ID.eq(processingReportId))
+                                .and(ACTION_LOG.REPORT_ID.eq(childReportId))
                                 .and(ACTION_LOG.TYPE.eq(ActionLogType.warning))
                         ).fetchInto(DetailedActionLog::class.java)
                     }
-                    actionResponses[processingReportId] = ret
-                } else actionResponses[processingReportId] = null
+                    actionResponses[childReportId] = ret
+                } else actionResponses[childReportId] = null
             }
 
             return actionResponses
         }
 
+        /**
+         * Returns the count of item lineages for the parent [reportId] passed in for the [action] specified.
+         * If a [receivingOrgSvc] or [receivingOrg] are specified, only the lineages that match those values will
+         * be counted. If this is looking for Universal Pipeline lineage count, use a different query due to how
+         * parent/child relationships work in lineage
+         * @return Count of matching records.
+         */
         fun itemLineageCountQuery(
             txn: DataAccessTransaction,
             reportId: ReportId,
             receivingOrgSvc: String? = null,
             receivingOrg: String? = null,
             action: TaskAction,
+            isUniversalPipeline: Boolean
         ): Int? {
             val ctx = DSL.using(txn)
-            val sql = """select count(*)
+            return if (isUniversalPipeline) {
+                val sql = """
```

Comment:
Thank you for letting me rubber duck with you on this, it actually makes sense when it definitely would not have otherwise.